File: app/src/main/java/kapil/circularlayoutmanager/RecyclerViewAdapter.java
package kapil.circularlayoutmanager;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

/**
 * Adapter for recycler view.
 */

class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder> {

    private Context context;
    private List<Model> list;

    RecyclerViewAdapter(Context context, List<Model> list) {
        this.context = context;
        this.list = list;
    }

    @Override
    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new MyViewHolder(LayoutInflater.from(context).inflate(R.layout.list_row, parent, false));
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        holder.event.setText(list.get(position).getEvent());
        holder.timings.setText(list.get(position).getTimings());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    /**
     * View Holder for recycler view.
     */

    class MyViewHolder extends RecyclerView.ViewHolder {
        TextView event, timings;

        MyViewHolder(View itemView) {
            super(itemView);
            event = (TextView) itemView.findViewById(R.id.event);
            timings = (TextView) itemView.findViewById(R.id.timings);
        }
    }
}

File: app/src/main/java/kapil/circularlayoutmanager/RecyclerViewAdapter.kt
package kapil.circularlayoutmanager

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import kapil.circularlayoutmanager.RecyclerViewAdapter.MyViewHolder
import kotlinx.android.synthetic.main.list_row.view.*


class RecyclerViewAdapter : ListAdapter<Model, MyViewHolder>(DIFF_CALLBACK) {

    var onItemClickListener: (Model.() -> Unit)? = null

    companion object {
        private val DIFF_CALLBACK = object : DiffUtil.ItemCallback<Model>() {
            override fun areItemsTheSame(oldItem: Model, newItem: Model) = oldItem.id == newItem.id
            override fun areContentsTheSame(oldItem: Model, newItem: Model) = oldItem == newItem
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder =
        MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.list_row, parent, false))

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) =
        holder.bindTo(getItem(position)!!)

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private val eventView: TextView = itemView.event
        private val timingsView: TextView = itemView.timings

        fun bindTo(model: Model) {
            eventView.text = model.event
            timingsView.text = model.timings
            itemView.setOnClickListener { onItemClickListener?.invoke(model) }
        }
    }
}